public/app/features/search/hooks/useDashboardSearch.ts

```diff
@@ -18,9 +19,27 @@ export const useDashboardSearch = (query: DashboardQuery, onCloseSearch: () => v
     dispatch,
   } = useSearch<DashboardsSearchState>(query, reducer, { queryParsing: true });
 
-  useEffect(() => {
-    reportDashboardListViewed('dashboard_search', showPreviews, previewFeatureEnabled, query.layout);
-  }, [showPreviews, previewFeatureEnabled, query.layout]);
+  useDebounce(
+    () => {
+      reportDashboardListViewed('dashboard_search', showPreviews, previewFeatureEnabled, {
+        layout: query.layout,
+        starred: query.starred,
+        sortValue: query.sort?.value,
+        query: query.query,
+        tagCount: query.tag?.length,
```

Comment:
Nit: I see this code in 3 places now - any chance for a util function that will get those properties given a query?

Reply:
will extract in the next PR :+1: